paddle/fluid/operators/reader/create_multi_pass_reader_op.cc

```diff
@@ -22,27 +22,31 @@ namespace reader {
 class MultiPassReader : public framework::DecoratedReader {
  public:
   MultiPassReader(ReaderBase* reader, int pass_num)
-      : DecoratedReader(reader), pass_num_(pass_num), pass_count_(0) {}
+      : DecoratedReader(reader), pass_num_(pass_num), cur_pass_(0) {}
 
-  void ReadNext(std::vector<framework::LoDTensor>* out) override {
-    reader_->ReadNext(out);
-    if (out->empty()) {
-      ++pass_count_;
-      if (pass_count_ < pass_num_) {
+  void ReInit() override {
+    cur_pass_ = 0;
+    reader_->ReInit();
+  }
+
+  LoDTensorListPtr ReadNext() override {
+    LoDTensorListPtr res = nullptr;
+    while (true) {
+      res = ReadNext();
```

Comment:
How about throwing an exception if `res` is still `nullptr` after invoking `ReInit`?